scripts/generate_pkcs7_tests.py

```diff
@@ -37,7 +39,15 @@ class TestData:
     Take in test_suite_pkcs7.data file.
     Allow for new tests to be added.
     """
-    mandatory_dep = "MBEDTLS_MD_CAN_SHA256"
+    # In Mbed TLS 3.6, PKCS7 code belongs to the legacy domain
+    # (see transition-guards.md for more information) thus use
+    # MBEDTLS_MD_CAN_SHA256 dependency symbol instead of PSA_WANT_ALG_SHA_256.
+    mandatory_dep = "MBEDTLS_MD_CAN_SHA256" if is_mbedtls_3_6() else "PSA_WANT_ALG_SHA_256"
+
+    # OR helper function but it does not help with the difficulty to decide
```

Comment:
For the choice of having a helper function or not: I think it mainly depends of what common patterns we have, how many instances of these patterns, and whether we're likely to have more instances in the future. My impression is:

* We have a small number of patterns. At least hashes, maybe PK algorithms?
* The number of instances is small, but larger than 1.
* There's a significant chance that we'll add more instances during the lifetime of 3.6, as we improve test coverage for X.509 and TLS.
* Furthermore the people adding these instances may be unfamiliar with 3.6 complexities, so we should make it discoverable. An optional argument on the PSA-related function is more discoverable than a pattern used in some other test script.

So I lean towards auxiliary functions.